lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -1914,6 +1902,40 @@ void GrandCentralPass::runOnOperation() {
                        "(including companion):\n";
               });
 
+              if (companionMode == CompanionMode::Drop) {
+                // Delete the instance if companions are disabled.
+                for (auto [index, port] :
+                     llvm::enumerate(instance->getResults())) {
+                  if (instance->getPortDirection(index) != Direction::In) {
+                    op.emitOpError()
+                        << "companion instance cannot have output ports";
+                    goto FModuleOp_error;
```

Comment:
This is a good check! This is common to all companion modules. Can this be moved out to a distinct check with a narrow test of this?